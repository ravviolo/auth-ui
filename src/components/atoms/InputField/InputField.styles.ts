import styled from 'styled-components';

import { colors } from 'theme';

export const Field = styled.input`
  width: 100%;
  padding: 0.5em;
  border-radius: 5px;
  border: 1px solid ${colors.textLight};
  cursor: pointer;
  transition: border-color 0.3s ease;

  &:hover {
    border-color: ${colors.text};
  }
  &:focus {
    border-color: ${colors.text};
    outline: none;
  }
`;
