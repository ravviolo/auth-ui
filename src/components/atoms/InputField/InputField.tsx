interface Props {
  name: string;
  id: string;
  type: React.HTMLInputTypeAttribute;
}

export const InputField = React.forwardRef<HTMLInputElement, Props>(
  ({ id, name, type }, ref) => {
    return <input ref={ref} id={id} name={name} type={type} />;
  }
);

InputField.displayName = 'InputField';
