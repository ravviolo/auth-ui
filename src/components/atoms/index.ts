export { Button } from './Button/Button';
export { Checkbox } from './Checkbox/Checkbox';
export { Divider } from './Divider/Divider';
export { Heading } from './Heading/Heading';
export { IconButton } from './IconButton/IconButton';
export { InputField } from './InputField/InputField';
export { InputLabel } from './InputLabel/InputLabel';
export { TextButton } from './TextButton/TextButton';
