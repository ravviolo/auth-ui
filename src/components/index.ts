export * from './templates/LoginTemplate/LoginTemplate';
export * from './templates/SignUpTemplate/SignUpTemplate';
