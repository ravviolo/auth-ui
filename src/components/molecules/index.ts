export { AuthFooter } from './AuthFooter/AuthFooter';
export { AuthHeader } from './AuthHeader/AuthHeader';
export { AuthSocials } from './AuthSocials/AuthSocials';
export { EmailInput } from './EmailInput/EmailInput';
export { PasswordInput } from './PasswordInput/PasswordInput';
