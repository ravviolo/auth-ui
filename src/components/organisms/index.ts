export * from './LoginForm/LoginForm';
export * from './SignUpForm/SignUpForm';
