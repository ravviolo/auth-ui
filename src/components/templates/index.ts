export * from './LoginTemplate/LoginTemplate';
export * from './SignUpTemplate/SignUpTemplate';
