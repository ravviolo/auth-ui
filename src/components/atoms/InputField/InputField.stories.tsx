import { ComponentMeta, ComponentStory } from '@storybook/react';

import { InputField } from 'components/atoms';

export default {
  title: 'Atoms/InputField',
  component: InputField,
  args: {
    id: 'testField',
    name: 'testField',
  },
} as ComponentMeta<typeof InputField>;

const Template: ComponentStory<typeof InputField> = (args) => (
  <InputField {...args} />
);

export const Text = Template.bind({});
Text.args = {
  type: 'text',
};

export const Password = Template.bind({});
Password.args = {
  type: 'password',
};
