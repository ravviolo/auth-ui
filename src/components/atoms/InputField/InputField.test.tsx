import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { create } from 'react-test-renderer';

import { InputField } from 'components/atoms';

import { Props as InputFieldProps } from './InputField';

describe('InputField', () => {
  const props: InputFieldProps = {
    id: 'test-input-field',
    name: 'test-input-field',
    testId: 'test-input-field',
    type: 'text',
  };

  it('renders initially as empty input', () => {
    render(<InputField {...props} />);

    const inputField = screen.getByTestId('test-input-field');

    expect(inputField).toHaveValue('');
  });

  it('correctly assigns id, name and type values received via props', () => {
    render(<InputField {...props} />);

    const inputField = screen.getByTestId('test-input-field');

    expect(inputField).toHaveAttribute('name', 'test-input-field');
    expect(inputField).toHaveAttribute('id', 'test-input-field');
    expect(inputField).toHaveAttribute('type', 'text');
  });

  it('displays entered text', async () => {
    render(<InputField {...props} />);

    const inputField = screen.getByTestId('test-input-field');

    await userEvent.type(inputField, 'Testing input field');

    expect(inputField).toHaveValue('Testing input field');
  });

  it('displays part of entered text after deleting some characters', async () => {
    render(<InputField {...props} />);

    // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
    const inputField = screen.getByTestId(
      'test-input-field'
    ) as HTMLInputElement;

    await userEvent.type(inputField, 'Testing input field');
    inputField.setSelectionRange(7, 19);
    await userEvent.keyboard('{backspace}');

    expect(inputField).toHaveValue('Testing');
  });

  it('matches snapshot', () => {
    const tree = create(<InputField {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
