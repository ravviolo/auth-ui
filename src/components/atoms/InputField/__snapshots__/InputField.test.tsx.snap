// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField matches snapshot 1`] = `
<input
  className="sc-ftvSup dLJLzu"
  data-testid="test-input-field"
  id="test-input-field"
  name="test-input-field"
  type="text"
/>
`;
